src/mobu/storage/git.py

```diff
@@ -0,0 +1,73 @@
+"""A very simple async replacement for GitPython, which has been
+abandoned and is distinctly not thread- or async-safe.
+"""
+
+from pathlib import Path
+
+from structlog.stdlib import BoundLogger
+
+from ..models.user import AuthenticatedUser
+from .process import Process
+
+
+class Git:
+    """A very basic async Git client based on asyncio.subprocess."""
+
+    def __init__(
+        self,
+        *,
+        repo: Path | None = None,
+        user: AuthenticatedUser | None = None,
+        logger: BoundLogger | None = None,
+    ) -> None:
+        self._user = user
+        if self._user is None:
+            self._user = AuthenticatedUser(
+                username="gituser",
+                uidnumber=60001,
+                gidnumber=60001,
+                scopes=[],
+                token="dummy token",
+            )
+        self._logger = logger
+        self._repo = repo
+        self._process = Process(user=self._user, logger=self._logger)
+
+    def set_repo(self, repo: Path) -> None:
+        self._repo = repo
```

Comment:
At first glance, the `set_repo` pattern feels like unnecessary state manipulation. Why not create a new `Git` object for every repo path? It looks like a lightweight object.

Reply:
Done.  The short answer is, it's a fair bit of ceremony each time.  But it wasn't too bad.